### ISSUE TITLE
[FEATURE] Support for Muon optimizer and WSD learning rate schedule

### ISSUE THREAD

**Issue by randydl** [baidu-baige/LoongForge#66]
Hi team,

Thanks for this great framework! Since it’s built on Megatron-Core, which already supports Muon, I’d like to ask:

- **Muon optimizer**: Is it directly usable in LoongForge, or does it need extra configuration?
- **WSD (Warmup-Stable-Decay) scheduler**: Is this learning rate schedule supported?

Appreciate your time!

**Comment by Zachary-wW** [comment]
Thanks for your interest!

LoongForge uses our patched Megatron fork, [Loong-Megatron](https://github.com/baidu-baige/Loong-Megatron), which is based on upstream Megatron-Core v0.15.0 (`core_v0.15.0`) with some extra modifications.

**For WSD**: yes, this is already supported in upstream Megatron-Core v0.15.0, so it is also available in LoongForge. You can enable it with Megatron's scheduler args, for example:

```bash
--lr-decay-style WSD \
--lr-wsd-decay-iters <decay_iters> \
--lr-wsd-decay-style exponential
```

The related args are here: [`--lr-decay-style WSD` / `--lr-wsd-decay-*`](https://github.com/baidu-baige/Loong-Megatron/blob/loong-main/core_v0.15.0/megatron/training/arguments.py#L2459-L2474).

**Muon Optimizer**: it's a bit different. We ported a Kimi-style Muon implementation into Loong-Megatron, mainly following [NVIDIA/Megatron-LM#1428](https://github.com/NVIDIA/Megatron-LM/pull/1428), and added some fixes and extensions on top of it, including distributed optimizer and optimizer CPU offload support.

In LoongForge, Muon can be enabled with:

```bash
--optimizer muon
```

Some Muon-specific knobs are also available, such as `--muon-matched-adamw-rms`, `--muon-momentum`, `--muon-ns-steps`, and `--no-muon-nesterov`. See: [Muon optimizer args](https://github.com/baidu-baige/Loong-Megatron/blob/loong-main/core_v0.15.0/megatron/training/arguments.py#L2000-L2008).

**Comment by randydl** [comment]
Many thanks for clearing up my questions and sharing the configuration details. It’s really helpful for my work. Hope this excellent project keeps evolving and getting better all the time.
